rules_doc_generator/model/section.py

```diff
@@ -267,10 +267,12 @@ def to_html(self, config: Config, id_map: RefDict) -> str:
     if self.snippet:
       result += f'<p class="Snippet">{self.snippet.to_html(config, id_map)}</p>'
     result += '<ol class="Rules">'
-    for elem in self.section_elements:
+    for n, elem in enumerate(self.section_elements):
+      rulenr:str = f'{id_map[self.id].reference}.{n+1}.'
+
       match elem:
-        case Rule(): result += f'<li class="Rule" id="{elem.id}">{elem.to_html(config, id_map)}</li>'
-        case SubSection(): result += f'<li class="Rule" id="{elem.id}">{elem.to_html(config, id_map)}</li>'
+        case Rule():       result += f'<li class="Rule" id="{elem.id}"><a class="RuleAnchor" href="#{elem.id}"></a><a class="RuleLink" href="?r={elem.id}">{rulenr}</a>{elem.to_html(config, id_map)}</li>'
```

Comment:
Why is the `RuleAnchor` element needed?

Reply:
From Discord:
Good question.. I forgot to remove that when I switched over to using scrollIntoView instead of anchors
Though I might still keep it in as a non-js alternative. Have the links point to the anchor in the html, but override that with js if scripts are enabled